go/vt/vtctl/workflow/server.go

```diff
@@ -1720,6 +1721,15 @@ func (s *Server) MoveTablesComplete(ctx context.Context, req *vtctldatapb.MoveTa
 		return nil, err
 	}
 
+	// Lock the workflow while we complete it.
+	lockName := fmt.Sprintf("%s/%s", ts.TargetKeyspaceName(), ts.WorkflowName())
+	ctx, workflowUnlock, lockErr := s.ts.LockName(ctx, lockName, "MoveTablesComplete")
+	if lockErr != nil {
+		ts.Logger().Errorf("Locking the workflow %s failed: %v", lockName, lockErr)
+		return nil, vterrors.Wrapf(lockErr, "failed to lock the %s workflow", lockName)
+	}
+	defer workflowUnlock(&err)
+
```

Comment:
The locking code seems to have been moved. Is there a consistent rule that we now follow for where we lock? We seem to call each lower level function (dropSources / DropTargets / DeleteTenantData) only in one place, so in practice we acquire/release the lock once regardless of where we do it.

Reply:
The keyspace locks wrench not moved. The workflow locks were moved so that the workflow is locked while the larger operation is happening. For example we hold it throughout the `WorkflowDelete` operation, which *ends* with deleting the workflow (and us releasing the workflow lock at the end). The keyspace locks we take only when needed and release ASAP as there's contention for those (e.g. schema changes).